.github/workflows/deploy-preview.yml

```diff
@@ -89,7 +89,7 @@ jobs:
         id: get_branch_name
         run: |
           BRANCH_NAME=${{ github.event.pull_request.head.ref }}
-          SANITIZED_BRANCH_NAME=$(echo "$BRANCH_NAME" | sed 's/[^a-zA-Z0-9]/-/g')
+          SANITIZED_BRANCH_NAME=$(echo "$BRANCH_NAME" | sed 's/[^a-zA-Z0-9._]/-/g')
```

Comment:
If i'm not mistaken, this would allow dots in the branch name. Is this used in the preview url? wouldn't it be an issue to have a dot in it?

Reply:
Yes, this allows dots, this is for the preview to preserve version names added to the branch when creating a pr

Reply:
I'm pretty sure SST will sanitise the stage name before using it as part of the URL.  It seems to convert all invalid characters to hyphens as well.  So yeah, maybe we should just do that upfront anyway?

Reply:
Yeah, we need the stage name to do a couple more things, generating one especially a sanitised one will help give a consistent name we can use.